app/src/main/java/io/gnosis/safe/ui/assets/coins/CoinsFragment.kt

```diff
@@ -51,6 +52,16 @@ class CoinsFragment : BaseViewBindingFragment<FragmentCoinsBinding>() {
                         when (action) {
                             is BaseStateViewModel.ViewAction.ShowError -> handleError(action.error)
                             is UpdateBalances -> adapter.setItems(action.newBalances)
+                            is Connectivity -> {
```

Comment:
This would have to be duplicated everywhere.

Reply:
Maybe that's not that terrible... but still. We should find a solution that doesn't mix responsibilities like this.

Reply:
You duplicate this only if you need to react to connectivity changes. All UI actions are explicitly listed for a screen, so I think this kind of duplication is good.

Reply:
I agree with that. This would be good duplication